frontend/src/components/ui/BuyMenu.jsx

```diff
@@ -215,6 +221,31 @@ export default function BuyMenu({game}) {
         )
     });
 
+    const undoBuild = () => {
+        if (!undoStack.length) return;
+        deselectAll();
+        const {x, y, z, label, price, tileType} = undoStack.at(-1);
+        
+        switch (label) {
+            case 'BuildTower':
+                game.removeTower(x, y);
+                gameMap.removeTower(x, y);
+                break;
+            case 'Fill':
+                gameMap.removeTile(x, y, z);
+                break;
+            case 'Excavate':
+                gameMap.addTile(x, y, z, new Tile(x, y, z, tileType));
+                break;
+            default:
+                console.error("Unknown undo label", label);
+        }
+
+        game.gold += price;
+        setUndoStack((prevStack) => prevStack.slice(0, -1));
+        game.setPath();
```

Comment:
We should make this `game.refreshPath()` or something like that instead. That would be a little bit more clear I think.